quickjs-libc.c

```diff
@@ -479,7 +488,54 @@ typedef JSModuleDef *(JSInitModuleFunc)(JSContext *ctx,
                                         const char *module_name);
 
 
-#if defined(_WIN32) || defined(__wasi__)
+#if defined(_WIN32)
+static JSModuleDef *js_module_loader_so(JSContext *ctx,
+                                        const char *module_name)
+{
+    JSModuleDef *m;
+    HINSTANCE hd;
+    JSInitModuleFunc *init;
+    char *filename = NULL;
+    int len = strlen(module_name);
+    if (len > 2 &&
+        ((module_name[0] >= 'A' && module_name[0] <= 'Z') ||
+            (module_name[0] >= 'a' && module_name[0] <= 'z')) &&
+        module_name[1] == ':') {
+        filename = (char *)module_name;
+    } else if (len > 2 && module_name[0] != '.' && (module_name[1]!= '/' || module_name[1] == '\\')) {
+        filename = js_malloc(ctx, strlen(module_name) + 2 + 1);
+        if (!filename)
+            return NULL;
+        strcpy(filename, "./");
+        strcpy(filename + 2, module_name);
+    }
+    {
+        hd = LoadLibraryA(filename);
+        if (filename!= module_name)
+            js_free(ctx, filename);
+        if (hd == NULL) {
+            JS_ThrowReferenceError(ctx, "js_load_module '%s' error: %lu",
+                                module_name, GetLastError());
+            goto fail;
+        }
+    }
+    init = (JSInitModuleFunc *)(void *)GetProcAddress(hd, "js_init_module");
+    if (!init) {
+        JS_ThrowReferenceError(ctx, "js_init_module '%s' not found: %lu",
+                               module_name, GetLastError());
```

Comment:
It'd be nicer to use FormatMessageA() to get a human-readable error message ([example](https://github.com/libuv/libuv/blob/58dfb6c89ba3deb23301ddbdd4a8769f4c767dff/src/win/dl.c#L105-L109))

Reply:
@bnoordhuis 
FormatMessageA error message fetching is a bit complicated, whether to consider a separate method to get it
```c
struct JSContext {
    char* errmsg;
}
static int js_error(JSContext *ctx) {
    if (ctx->errmsg) {
    LocalFree(ctx->errmsg);
    ctx->errmsg = NULL;
  }
  // FormatMessageA ……
}
```

Reply:
@bnoordhuis Doing it The Right Way (TM) seems a bit involved: https://github.com/libuv/libuv/blob/0a00e80c3686b93eccb9a801954e86bd7d7fe6ab/src/win/dl.c#L92 perhaps we can simplify here?